pytensor/scalar/basic.py

```diff
@@ -1142,12 +1142,15 @@ def output_types(self, types):
 
     def perform(self, node, inputs, output_storage):
         if self.nout == 1:
-            output_storage[0][0] = self.impl(*inputs)
+            output_storage[0][0] = np.asarray(
+                self.impl(*inputs),
+                dtype=node.outputs[0].dtype,
+            )
         else:
             variables = from_return_values(self.impl(*inputs))
             assert len(variables) == len(output_storage)
-            for storage, variable in zip(output_storage, variables):
-                storage[0] = variable
+            for out, storage, variable in zip(node.outputs, output_storage, variables):
+                storage[0] = np.asarray(variable, dtype=out.dtype)
```

Comment:
Outputs should be scalars not arrays

Reply:
Can't you just call `.astype(out.dtype)`?

Reply:
They're scalar arrays, sometimes function returns floats and astype method is missing.

Reply:
Scalar arrays are not the same. These functions should return non array scalars

Reply:
What are those?

Reply:
`np.float64(5.0), np int32(3)` and so on